File: 0066-plus-one/0066-plus-one.cpp
class Solution
{
public:
    vector<int> plusOne(vector<int> &digits)
    {
        int lastElement = digits.size() - 1;
        int carry = 1;
        for (int i = lastElement; i > -1; i--)
        {
            int res = digits[i] +carry;
            carry = res / 10;
            res = res % 10;
            digits[i] = res;
            if (carry == 0)
                return digits;
        }

        vector<int> v(digits.size() + 1);
        v[0] = carry;
        for (int i = 1; i < digits.size(); i++)
        {
            v[i] = digits[i - 1];
        }

        return v;
    }
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;  
        List<Integer> result = new ArrayList<>();
        
        for (int i = digits.length - 1; i >= 0; i--) {
            int temp = carry + digits[i];
            result.add(temp % 10);  
            carry = temp / 10;       
        }
        
        if (carry > 0) {
            result.add(carry);
        }
        
        
        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(result.size() - 1 - i);
        }
        
        return answer;
    }
}